advisor-go/src/quixo/simulation/simulation.go

```diff
@@ -16,35 +16,54 @@ func GetBestMoveForPlayer(board game.Board) Move {
 	return getBestMove(board, false)
 }
 
-// GetWorstMoveForPlayer return the move that will benefit the most to the opponent player
+// GetWorstMoveForPlayer return the move that will benefit the most for the opponent player
 func GetWorstMoveForPlayer(board game.Board) Move {
 	return getBestMove(board, true)
 }
 
+// GetMoveThatMakeMyOpponentLose return the move that will be the worst for the opponent
+func GetMoveThatMakeMyOpponentLose(board game.Board) Move {
+	isSmallerThan := func(int1 int, int2 int) bool { return int1 < int2 }
+	if hasCubeSelected(board) {
+		return getMoveWithCubeSelected(board, true, isSmallerThan)
+	}
+	return getMoveWithNoCubeSelected(board, true, isSmallerThan)
+}
+
 func getBestMove(board game.Board, forOpponentPlayer bool) Move {
+	isGreaterThan := func(int1 int, int2 int) bool { return int1 > int2 }
+
 	if hasCubeSelected(board) {
-		return getBestMoveWithCubeSelected(board, forOpponentPlayer)
+		return getMoveWithCubeSelected(board, forOpponentPlayer, isGreaterThan)
 	}
-	return getBestMoveWithNoCubeSelected(board, forOpponentPlayer)
+	return getMoveWithNoCubeSelected(board, forOpponentPlayer, isGreaterThan)
 }
 
-func getBestMoveWithCubeSelected(board game.Board, forOpponentPlayer bool) Move {
-	destination, _ := getBestDestinationWithScore(board, board.SelectedCube, forOpponentPlayer)
+func getMoveWithCubeSelected(
+	board game.Board,
+	forOpponentPlayer bool,
+	comparator func(int, int) bool,
+) Move {
+	destination, _ := getBestDestinationWithScore(board, board.SelectedCube, forOpponentPlayer, comparator)
 	return Move{
 		board.SelectedCube.Coords,
 		destination,
 	}
 }
 
-func getBestMoveWithNoCubeSelected(board game.Board, forOpponentPlayer bool) Move {
+func getMoveWithNoCubeSelected(
+	board game.Board,
+	forOpponentPlayer bool,
+	comparator func(int, int) bool,
+) Move {
 	movables := game.GetMovablesCubes(board)
 	bestMove := Move{}
-	maxScore := -1
+	bestScore := 0
 	for i := 0; i < len(movables); i++ {
-		destination, score := getBestDestinationWithScore(board, movables[i], forOpponentPlayer)
+		destination, score := getBestDestinationWithScore(board, movables[i], forOpponentPlayer, comparator)
 
-		if score > maxScore {
-			maxScore = score
+		if comparator(score, bestScore) || i == 0 {
```

Comment:
why ? `|| 1== 0` `comparator` doesn't always return a boolean ?

Reply:
It's `i == 0`, so bestScore is assigned the value of score on the first loop. I didn't found a best way to do that.

Reply:
Damned, i'm old :cry: Yep, I had read 1 and not i